### PR TITLE
1611 day to be enterable for expiry date

### LINKED ISSUE #1611: Day to be enterable for expiry date
## Is your feature request related to a problem? Please describe 👀

@richardmoizeau noticed that `day` is not enterable for expiry date

## Describe the solution you'd like 🎁

Allow 'day' to be entered in expiry (only month and year are currently enterable). Still keep the helper calendar to be just month and year (as it is in mSupply

### Describe alternatives you've considered 💭

Keeping as is, @GaryWilletts mentioned that not all expiry dates are end of month

### PR DESCRIPTION
**Pull request by roxy-dao**
Fixes #1611

# 👩🏻‍💻 What does this PR do?
Allows the user to enter a date with the keyboard but keep selection as MM/yyyy, and display the expiry date in full. If the user uses the selection, then the default day should be the end of that month. 

Opted to force desktop view so that the day can be enterable in Android as well.

Jun-27-2024 17-54-53

# 🧪 Testing

- [ ] Create a new stocktake 
- [ ] Go to a line and try typing an expiry day, save.
- [ ] Go back to line and change date through selection instead of typing
- [ ] Day should change to last day of whatever month you selection.
- [ ] 2. Go to Stock page. Repeat above instructions.

# 📃 Documentation

- [ ] **Part of an epic**: documentation will be completed for the feature as a whole
- [ ] **No documentation required**: no user facing changes or a bug fix which isn't a change in behaviour
- [x] **These areas should be updated or checked**: 
  1. Anywhere displaying expiry date
  2.

### CODE REVIEW

File: client/packages/common/src/ui/components/inputs/DatePickers/BaseDatePickerInput/BaseDatePickerInput.tsx [hunk 1]
```diff
@@ -58,7 +58,7 @@ export const BaseDatePickerInput: FC<
   const t = useTranslation();
 
   return (
-    <DatePicker
+    <DesktopDatePicker
```
Comment (roxy-dao) [inline client/packages/common/src/ui/components/inputs/DatePickers/BaseDatePickerInput/BaseDatePickerInput.tsx:61]: Forcing to allow user to enter day in Android

File: client/packages/common/src/ui/layout/tables/utils/ColumnDefinitionSetBuilder.ts [hunk 2]
```diff
@@ -84,12 +83,7 @@ const getColumnLookup = <T extends RecordWithId>(): Record<
     key: 'expiryDate',
     label: 'label.expiry',
     width: 110,
-    formatter: dateString => {
-      if (dateString === '[multiple]') return '[multiple]';
-      return dateString
-        ? Formatter.expiryDate(new Date(dateString as string)) || ''
-        : '';
-    },
+    format: ColumnFormat.Date,
```
Comment (roxy-dao) [inline client/packages/common/src/ui/layout/tables/utils/ColumnDefinitionSetBuilder.ts:86]: Displaying in full for now until confirmation by @mark-prins

Comment (mark-prins) [reply client/packages/common/src/ui/layout/tables/utils/ColumnDefinitionSetBuilder.ts:86]: yes, in full thanks 🙏

**Review by lache-melvin — CHANGES_REQUESTED** [review]
🔥 

Adding new dates looks good but when I try type to edit existing date, can edit day but then it gets set to end of month when I try edit the month?

https://github.com/msupply-foundation/open-msupply/assets/55115239/d10143c7-12d8-48fd-a7f0-029950ab533e

**Comment by CarlosNZ** [comment]
> Adding new dates looks good but when I try type to edit existing date, can edit day but then it gets set to end of month when I try edit the month?

@lache-melvin -- @roxy-dao asked me to have a look at this, as being able to detect whether the date has been updated via the Picker or via keyboard input was proving tricky. It's true, there's no way to tell just based on events. So I've come up with this solution: 95d33590e

I'll leave it to @roxy-dao to do a final check and merge if we're all happy with this fix.


(@roxy-dao -- I also removed a `useState` as it didn't seem necessary (can get same data from `value`, plus the `setDate` wasn't guaranteed to be updated in time for the next check that uses it, which is often a source of bugs) )

**Comment by roxy-dao** [comment]
> > Adding new dates looks good but when I try type to edit existing date, can edit day but then it gets set to end of month when I try edit the month?
> 
> @lache-melvin -- @roxy-dao asked me to have a look at this, as being able to detect whether the date has been updated via the Picker or via keyboard input was proving tricky. It's true, there's no way to tell just based on events. So I've come up with this solution: [95d3359](https://github.com/msupply-foundation/open-msupply/commit/95d33590e3e4b6083c20ff8918e25698fb06e372)
> 
> I'll leave it to @roxy-dao to do a final check and merge if we're all happy with this fix.
> 
> (@roxy-dao -- I also removed a `useState` as it didn't seem necessary (can get same data from `value`, plus the `setDate` wasn't guaranteed to be updated in time for the next check that uses it, which is often a source of bugs) )

Yup! Seems good to me thank you 🙏🏻 and removing state is good call (:

File: server/repository/src/migrations/v2_00_00/mod.rs [hunk 3]
```diff
@@ -52,7 +52,7 @@ impl Migration for V2_00_00 {
         run_without_change_log_updates(connection, |_| {
             stocktake_line_add_item_name::migrate(connection)
         })?;
-        name_created_datetime::migrate(connection)?;
+        name_deleted_datetime::migrate(connection)?;
```
Comment (roxy-dao) [inline server/repository/src/migrations/v2_00_00/mod.rs:55]: not sure why this was created when it's creating the deleted column 🤷🏻

**Review by lache-melvin — APPROVED** [review]
Looks awesome team thank you!!

**Pull request merged**